R/data-data-utils.R

```diff
@@ -0,0 +1,75 @@
+#' Function runs the `code`, masks the `code` and creates `teal_data` object.
+#' @param input (`list`) containing inputs to be used in the `code`
+#' @inheritParams tm_teal_data
+#'
+#' @return `teal_data` object
+#'
+#' @export
+eval_and_mask <- function(data,
+                          code,
+                          input = list(),
+                          input_mask = list()) {
+  checkmate::assert_list(input)
+  if (inherits(input, "reactivevalues")) {
+    input <- shiny::reactiveValuesToList(input)
+  }
+
+  # evaluate code and substitute input
+  data <- teal.code::eval_code(data, .mask_code(code, args = input))
+
+  if (identical(ls(data@env), character(0))) {
```

Comment:
when I am passing incorrect credential it returned data is of class qenv.error but does not have env attribute. 

```
Browse[1]> ls(data@env)
Warning in ls(data@env) : ‘data@env’ converted to character string
Error in as.environment(pos) : 
  no item called "data@env" on the search list
Browse[1]> data
<qenv.error: Invalid credentials. 'pass' is the password 
 when evaluating qenv code:
open_conn(username = "erer", password = "rere")>
```

Reply:
Fixed (kind of)